utils/spdx/src/main/kotlin/Utils.kt

```diff
@@ -130,8 +130,18 @@ fun getLicenseTextReader(
 ): (() -> String)? {
     return if (id.startsWith(LICENSE_REF_PREFIX)) {
         getLicenseTextResource(id)?.let { { it.readText() } }
-            ?: addScanCodeLicenseTextsDir(licenseTextDirectories).firstNotNullOfOrNull {
-                getLicenseTextFile(id, it)?.let { file -> { file.readText() } }
+            ?: addScanCodeLicenseTextsDir(licenseTextDirectories).firstNotNullOfOrNull { dir ->
+                getLicenseTextFile(id, dir)?.let { file ->
+                    {
+                        val lines = file.readLines()
+
+                        // Remove any YAML front matter enclosed by "---" from ScanCode license files.
+                        val licenseLines = lines.takeUnless { it.first() == "---" }
+                            ?: lines.drop(1).dropWhile { it != "---" }.drop(1)
+
+                        licenseLines.joinToString("\n").trim()
+                    }
+                }
```

Comment:
Would it make sense to extract lines 134-144 into a functions e.g. `getScanCodeLicenseText(id, dir)`
and add a test which calls it for a file which has such YAML in the header, to show it works?

Reply:
Done. I've actually improved the algorithm a bit as part of adding the test.